### PR TITLE
feat: promisify debugger.sendCommand()

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change

This PR promisifies `debugger.sendCommand()`

/cc @codebytere

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes
Notes: Converted `debugger.sendCommand()` to return a Promise instead of taking a callback.

### CODE REVIEW

File: atom/browser/api/atom_api_debugger.cc [hunk 1]
```diff
@@ -61,23 +61,31 @@ void Debugger::DispatchProtocolMessage(DevToolsAgentHost* agent_host,
...
     Emit("message", method, params);
   } else {
-    auto send_command_callback = pending_requests_[id];
-    pending_requests_.erase(id);
-    if (send_command_callback.is_null())
+    auto it = pending_requests_.find(id);
+    if (it == pending_requests_.end())
       return;
+
+    auto promise = it->second;
```
[the remark this answers is not in the document]
Comment (miniak) [reply atom/browser/api/atom_api_debugger.cc]: added type

File: atom/browser/api/atom_api_debugger.cc [hunk 2]
```diff
@@ -160,7 +173,7 @@ void Debugger::ClearPendingRequests() {
   base::Value error_msg("target closed while handling command");
   error.SetKey("message", std::move(error_msg));
   for (const auto& it : pending_requests_)
-    it.second.Run(error, base::Value());
+    it.second->Reject(error);
```
[the remark this answers is not in the document]
Comment (miniak) [reply atom/browser/api/atom_api_debugger.cc]: replaced both `Reject` with `RejectWithErrorMessage`, modified tests for check for `Error` instances

File: spec/api-debugger-spec.js [hunk 3]
```diff
@@ -123,7 +138,24 @@ describe('debugger module', () => {
       w.webContents.debugger.sendCommand('Runtime.evaluate', params, callback)
     })
 
-    it('returns response when devtools is opened', done => {
+    it('returns response when devtools is opened', async () => {
+      w.webContents.loadURL('about:blank')
+      w.webContents.debugger.attach()
+
+      w.webContents.openDevTools()
+      await emittedOnce(w.webContents, 'devtools-opened')
```
Comment (nornagon) [inline spec/api-debugger-spec.js]: 🚨 FLAKE ALERT 🚨 

you need to call `emittedOnce` _before_ you call `openDevTools`. As written this is a race condition and will result in a flaky test.

File: spec/chromium-spec.js [hunk 4]
```diff
@@ -1515,11 +1515,7 @@ describe('font fallback', () => {
     try {
       await w.loadURL(`data:text/html,${html}`)
       w.webContents.debugger.attach()
-      const sendCommand = (...args) => new Promise((resolve, reject) => {
-        w.webContents.debugger.sendCommand(...args, (e, r) => {
-          if (e) { reject(e) } else { resolve(r) }
-        })
-      })
+      const sendCommand = (...args) => w.webContents.debugger.sendCommand(...args)
```
Comment (nornagon) [inline spec/chromium-spec.js]: Can probably just call `w.webContents.debugger.sendCommand` directly, or if that's too verbose then `const {debugger} = w.webContents; debugger.sendCommand(...)`

**Pull request merged**